root/static/scripts/edit/URLCleanup.js

```diff
@@ -2935,6 +2945,16 @@ const CLEANUPS: CleanupEntries = {
       return {result: false, target: ERROR_TARGETS.URL};
     },
   },
+  'melon': {
+    match: [new RegExp('^(https?://)?((www|m2)\\.)?melon\\.com/', 'i')],
+    restrict: [multiple(LINK_TYPES.downloadpurchase, LINK_TYPES.streamingpaid), LINK_TYPES.streamingpaid],
+    select: function (url, sourceType) {
+      return [LINK_TYPES.downloadpurchase.release, LINK_TYPES.streamingpaid.release];
+    },
+    clean: function (url) {
+      return url.replace(/^(?:https?:\/\/)?(?:(?:www|m2)\.)?melon\.com\/album\/(?:detail|music)\.htm\?albumId=(\d+)(?:[\/?#].*)?$/, 'https://www.melon.com/album/detail.htm?albumId=$1');
+    },
```

Comment:
Cleaning album URLs is fine. Would be worth cleaning artist URLs too?

Reply:
There currently are:
* 2390 album URLs
* 135 artist URLs
* 28 song URLs
* 1 video URLs
* and 6 hazardous URLs including search results
from Melon linked into the MusicBrainz database.

So albums are even more dominant here.

Reply:
Another question, how can I get those counts myself? Does that work on then production database too or do I need a full database dump?

Reply:
For rough searches you can use the [search API](https://musicbrainz.org/doc/MusicBrainz_API/Search). For example: https://musicbrainz.org/ws/2/url?query=url:*www.melon.com/*&fmt=json&offset=0&limit=1

For searches based on regular expression, it requires setting up a local database mirror which takes 40 GB at least.

Reply:
Here is a rough list of Melon URLs from the MBDB: https://gist.github.com/yvanzo/f649a94ae6426b07f2c0d0803cd95a4a